src/ToolbarService.php

```diff
@@ -105,6 +107,33 @@ public function isEnabled()
         return $force;
     }
 
+    /**
+     * Returns true if this applications is being executed on a domain with a TLD
+     * that is commonly associated with a production environment.
+     *
+     * @return bool
+     */
+    protected function isNotSuspiciouslyProduction()
+    {
+        $host = explode('.', parse_url('http://' . env('HTTP_HOST'), PHP_URL_HOST));
+        $first = current($host);
+        $isIP = is_numeric(implode('', $host));
+
+        if (count($host) === 1) {
+            return false;
+        }
+
+        if ($isIP && in_array($first, ['192', '10', '127'])) {
```

Comment:
My production environment at work has webservers on 10. addresses, as our public network terminates at the load balancer in front of the webservers. If we want to err on the cautious side we might not want to use this check.

Reply:
@markstory but the request host would not be the IP unless you had access to the local private network, that's why I think it is safe.

Reply:
Good point.